src/instruments/src/PFD/index.tsx

```diff
@@ -8,247 +10,218 @@ import { HeadingOfftape, HeadingTape } from './HeadingIndicator';
+import { getSimVar, setSimVar } from '../util.js';
 import { SmoothSin, LagFilter, RateLimiter } from './PFDUtils';
 import { DisplayUnit } from '../Common/displayUnit';
 import { render } from '../Common';
 import './style.scss';
 
-/* eslint-disable max-len */
-// eslint-disable-next-line react/prefer-stateless-function
-class PFD extends Component {
-    private displayIndex: number;
-
-    private deltaTime: number;
-
-    private GetDeltaTime: () => number;
-
-    private prevAirspeed: number;
-
-    private VLs: number;
-
-    private barTimer: number;
-
-    private smoothFactor: number;
-
-    private isAttExcessive: boolean;
-
-    private AirspeedAccFilter: LagFilter;
-
-    private AirspeedAccRateLimiter: RateLimiter;
-
-    private LSButtonPressed: boolean;
+export const PFD: React.FC = () => {
+    const [displayIndex] = useState(() => {
+        const url = document.getElementsByTagName('a32nx-pfd')[0].getAttribute('url');
+        return url ? parseInt(url.substring(url.length - 1), 10) : 0;
+    });
 
-    private failuresConsumer: FailuresConsumer;
+    const [failuresConsumer] = useState(() => {
+        const consumer = new FailuresConsumer('A32NX');
+        consumer.register(isCaptainSide(displayIndex) ? A320Failure.LeftPfdDisplay : A320Failure.RightPfdDisplay);
 
-    constructor(props: {}) {
-        super(props);
+        return consumer;
+    });
 
-        const url = document.getElementsByTagName('a32nx-pfd')[0].getAttribute('url');
-        if (url) {
-            this.displayIndex = parseInt(url.substring(url.length - 1), 10);
-        }
+    const [previousAirspeed, setPreviousAirspeed] = useState(0);
+    const [clampedAirspeed, setClampedAirspeed] = useState(0);
+    const [filteredAirspeedAcc, setfilteredAirspeedAcc] = useState(0);
 
-        this.deltaTime = 0;
-        this.GetDeltaTime = createDeltaTimeCalculator();
-        this.prevAirspeed = 0;
+    const [airspeedAccFilter] = useState(() => new LagFilter(1.2));
+    const [airspeedAccRateLimiter] = useState(() => new RateLimiter(1.2, -1.2));
 
-        this.VLs = 0;
+    const [isAttExcessive, setIsAttExcessive] = useState(false);
 
-        this.barTimer = 10;
+    const [lsButtonPressed, setLsButtonPressed] = useState(false);
 
-        this.smoothFactor = 0.5;
+    const [barTimer, setBarTimer] = useState(10);
+    const [showSpeedBars, setShowSpeedBars] = useState(true);
 
-        this.isAttExcessive = false;
+    const inertialReferenceSource = getSupplier(displayIndex, getSimVar('L:A32NX_ATT_HDG_SWITCHING_KNOB', 'Enum'));
+    const airDataReferenceSource = getSupplier(displayIndex, getSimVar('L:A32NX_AIR_DATA_SWITCHING_KNOB', 'Enum'));
+    const computedAirspeed = useArinc429Var(`L:A32NX_ADIRS_ADR_${airDataReferenceSource}_COMPUTED_AIRSPEED`);
 
-        this.AirspeedAccFilter = new LagFilter(1.2);
-        this.AirspeedAccRateLimiter = new RateLimiter(1.2, -1.2);
+    const isOnGround = getSimVar('SIM ON GROUND', 'Bool');
```

Comment:
Not directly relevant to this PR, but we actually have LGCIU inputs for this now based on gear compression, although I'm not sure those are exported as a SimVar already.